sql/catalog/lease: TestKVWriterMatchesIEWriter failed

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14475747?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14475747?buildTab=artifacts#/) on master @ [5c26990f19a8f83c0a4848cc4245ac1823cb376a](https://github.com/cockroachdb/cockroach/commits/5c26990f19a8f83c0a4848cc4245ac1823cb376a):


```
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 35 35 2d 30 32  |........'2255-02|
        	            	+   00000010  2d 30 36 20 30 35 3a 31  38 3a 35 39 2e 36 30 32  |-06 05:18:59.602|
        	            	+   00000020  38 39 33 27 5f 73 65 73  73 69 6f 6e 00 01 88     |893'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 08                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 35 37 2d 30 32  |........'2257-02|
        	            	+   00000010  2d 31 32 20 32 32 3a 30  31 3a 31 32 2e 36 38 37  |-12 22:01:12.687|
        	            	+   00000020  32 30 37 27 5f 73 65 73  73 69 6f 6e 00 01 88     |207'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) {
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 35 37 2d 30 32  |........'2257-02|
        	            	+   00000010  2d 31 32 20 32 32 3a 30  31 3a 31 32 2e 36 38 37  |-12 22:01:12.687|
        	            	+   00000020  32 30 37 27 5f 73 65 73  73 69 6f 6e 00 01 88     |207'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 00                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ }
        	            	+}
        	            	 
        	Test:       	TestKVWriterMatchesIEWriter/mode=SessionBasedOnly
    --- FAIL: TestKVWriterMatchesIEWriter/mode=SessionBasedOnly (10.46s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-36864

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14477863?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14477863?buildTab=artifacts#/) on master @ [ecedafa52fd732fa29052cdafd393a4df4496eff](https://github.com/cockroachdb/cockroach/commits/ecedafa52fd732fa29052cdafd393a4df4496eff):


```
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 30 34 2d 30 39  |........'2204-09|
        	            	+   00000010  2d 32 30 20 31 30 3a 31  33 3a 31 35 2e 32 31 39  |-20 10:13:15.219|
        	            	+   00000020  35 30 32 27 5f 73 65 73  73 69 6f 6e 00 01 88     |502'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 00                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 34 36 2d 30 34  |........'2246-04|
        	            	+   00000010  2d 30 35 20 31 31 3a 35  34 3a 33 38 2e 37 35 36  |-05 11:54:38.756|
        	            	+   00000020  31 30 35 27 5f 73 65 73  73 69 6f 6e 00 01 88     |105'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 0e                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 35 34 2d 30 36  |........'2254-06|
        	            	+   00000010  2d 31 37 20 32 33 3a 30  31 3a 30 38 2e 36 30 31  |-17 23:01:08.601|
        	            	+   00000020  35 32 31 27 5f 73 65 73  73 69 6f 6e 00 01 88     |521'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 04                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ }
        	            	+}
        	            	 
        	Test:       	TestKVWriterMatchesIEWriter/mode=SessionBasedOnly
    --- FAIL: TestKVWriterMatchesIEWriter/mode=SessionBasedOnly (11.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14482904?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14482904?buildTab=artifacts#/) on master @ [3060756a084cc437114d64265e79ace5720a7317](https://github.com/cockroachdb/cockroach/commits/3060756a084cc437114d64265e79ace5720a7317):


```
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 35 36 2d 30 34  |........'2256-04|
        	            	+   00000010  2d 30 35 20 32 31 3a 35  37 3a 30 30 2e 39 36 30  |-05 21:57:00.960|
        	            	+   00000020  35 39 38 27 5f 73 65 73  73 69 6f 6e 00 01 88     |598'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) {
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 35 36 2d 30 34  |........'2256-04|
        	            	+   00000010  2d 30 35 20 32 31 3a 35  37 3a 30 30 2e 39 36 30  |-05 21:57:00.960|
        	            	+   00000020  35 39 38 27 5f 73 65 73  73 69 6f 6e 00 01 88     |598'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 02                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 36 32 2d 30 31  |........'2262-01|
        	            	+   00000010  2d 32 31 20 31 34 3a 30  31 3a 35 35 2e 32 31 35  |-21 14:01:55.215|
        	            	+   00000020  30 37 34 27 5f 73 65 73  73 69 6f 6e 00 01 88     |074'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 00                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ }
        	            	+}
        	            	 
        	Test:       	TestKVWriterMatchesIEWriter/mode=SessionBasedOnly
    --- FAIL: TestKVWriterMatchesIEWriter/mode=SessionBasedOnly (10.14s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://tanzanite.cluster.engflow.com/invocations/default/003514ab-4fc5-42b8-8eac-d72066107901?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [3060756a084cc437114d64265e79ace5720a7317](https://github.com/cockroachdb/cockroach/commits/3060756a084cc437114d64265e79ace5720a7317):


```
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 34 32 2d 31 30  |........'2242-10|
        	            	+   00000010  2d 32 35 20 30 33 3a 32  36 3a 31 39 2e 30 32 32  |-25 03:26:19.022|
        	            	+   00000020  31 31 32 27 5f 73 65 73  73 69 6f 6e 00 01 88     |112'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 0c                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 34 35 2d 30 32  |........'2245-02|
        	            	+   00000010  2d 30 33 20 31 39 3a 34  32 3a 33 38 2e 35 34 35  |-03 19:42:38.545|
        	            	+   00000020  30 33 38 27 5f 73 65 73  73 69 6f 6e 00 01 88     |038'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 0a                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 35 37 2d 30 34  |........'2257-04|
        	            	+   00000010  2d 32 36 20 31 31 3a 34  39 3a 31 34 2e 34 38 35  |-26 11:49:14.485|
        	            	+   00000020  39 38 33 27 5f 73 65 73  73 69 6f 6e 00 01 88     |983'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 0e                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ }
        	            	+}
        	            	 
        	Test:       	TestKVWriterMatchesIEWriter/mode=SessionBasedOnly
    --- FAIL: TestKVWriterMatchesIEWriter/mode=SessionBasedOnly (22.33s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://tanzanite.cluster.engflow.com/invocations/default/613e74e8-563c-43e0-b135-d85141636ac3?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [3060756a084cc437114d64265e79ace5720a7317](https://github.com/cockroachdb/cockroach/commits/3060756a084cc437114d64265e79ace5720a7317):


```
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 32 36 2d 30 38  |........'2226-08|
        	            	+   00000010  2d 30 34 20 30 38 3a 33  33 3a 33 30 2e 31 34 30  |-04 08:33:30.140|
        	            	+   00000020  37 31 27 5f 73 65 73 73  69 6f 6e 00 01 88        |71'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 12                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=46) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 34 36 2d 30 38  |........'2246-08|
        	            	+   00000010  2d 31 33 20 32 30 3a 32  37 3a 31 36 2e 30 31 36  |-13 20:27:16.016|
        	            	+   00000020  37 37 27 5f 73 65 73 73  69 6f 6e 00 01 88        |77'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 08                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 34 39 2d 31 31  |........'2249-11|
        	            	+   00000010  2d 32 35 20 32 30 3a 32  36 3a 35 31 2e 34 34 37  |-25 20:26:51.447|
        	            	+   00000020  30 36 33 27 5f 73 65 73  73 69 6f 6e 00 01 88     |063'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 06                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ }
        	            	+}
        	            	 
        	Test:       	TestKVWriterMatchesIEWriter/mode=SessionBasedOnly
    --- FAIL: TestKVWriterMatchesIEWriter/mode=SessionBasedOnly (27.05s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14500393?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14500393?buildTab=artifacts#/) on master @ [8dcbdc44a95c85e1e126099d115fca3ce81b3df6](https://github.com/cockroachdb/cockroach/commits/8dcbdc44a95c85e1e126099d115fca3ce81b3df6):


```
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 32 39 2d 30 36  |........'2229-06|
        	            	+   00000010  2d 30 31 20 32 32 3a 30  35 3a 30 30 2e 34 38 34  |-01 22:05:00.484|
        	            	+   00000020  34 36 38 27 5f 73 65 73  73 69 6f 6e 00 01 88     |468'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 06                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 32 39 2d 31 32  |........'2229-12|
        	            	+   00000010  2d 31 36 20 30 33 3a 30  36 3a 30 30 2e 37 39 39  |-16 03:06:00.799|
        	            	+   00000020  34 35 31 27 5f 73 65 73  73 69 6f 6e 00 01 88     |451'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 06                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=47) {
        	            	+   00000000  8b 12 80 00 01 91 91 12  27 32 32 34 35 2d 31 30  |........'2245-10|
        	            	+   00000010  2d 32 35 20 30 34 3a 34  30 3a 34 35 2e 31 37 38  |-25 04:40:45.178|
        	            	+   00000020  30 35 33 27 5f 73 65 73  73 69 6f 6e 00 01 88     |053'_session...|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=7) {
        	            	+    00000000  00 00 00 00 0a 33 0a                              |.....3.|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0
        	            	+   }
        	            	+  }
        	            	+ }
        	            	+}
        	            	 
        	Test:       	TestKVWriterMatchesIEWriter/mode=SessionBasedOnly
    --- FAIL: TestKVWriterMatchesIEWriter/mode=SessionBasedOnly (10.79s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/catalog/lease/kv_writer_test.go
+++ b/pkg/sql/catalog/lease/kv_writer_test.go
@@ -92,9 +92,15 @@ func TestKVWriterMatchesIEWriter(t *testing.T) {
 			tdb := sqlutils.MakeSQLRunner(sqlDB)
 			schema := systemschema.LeaseTableSchema
 			if mode == SessionBasedOnly {
-				schema = systemschema.LeaseTableSchema_V24_1
+				schema = strings.Replace(systemschema.LeaseTableSchema_V24_1,
+					"exclude_data_from_backup = true",
+					"exclude_data_from_backup = false",
+					1)
+
 			}
 			makeTable := func(name string) (id descpb.ID) {
+				// Rewrite the schema and drop the exclude_data_from_backup from flag,
+				// since this will prevent export from working later on in the test.
 				tdb.Exec(t, strings.Replace(schema, "system.lease", name, 1))
 				tdb.QueryRow(t, "SELECT id FROM system.namespace WHERE name = $1", name).Scan(&id)
 				// Modifies the primary index IDs to line up with the session based
@@ -108,8 +114,7 @@ func TestKVWriterMatchesIEWriter(t *testing.T) {
 			}
 			lease1ID := makeTable("lease1")
 			lease2ID := makeTable("lease2")
-
-			ie := s.InternalExecutor().(isql.Executor)
+			ie := s.InternalDB().(isql.DB).Executor()
 			codec := s.Codec()
 			settingsWatcher := s.SettingsWatcher().(*settingswatcher.SettingsWatcher)
 			modeReader := &dummySessionModeReader{mode: mode}

